pallets/pallet-bank/src/lib.rs

```diff
@@ -167,6 +232,9 @@ pub mod module {
 
         fn on_finalize(_block_number: BlockNumberFor<T>) {
             Self::reap_accounts();
+            AccountWithUnlockedFund::<T>::take(&_block_number).iter().for_each(|(user, lock_id)|{
```

Comment:
nitpick: since the result from the "take" is thrown away, I think into_iter can be used here.